lib/rubocop/cop/grep/grep.rb

```diff
@@ -42,9 +42,10 @@ def on_new_investigation
           { line: line, column: column, length: length }
         end
 
-        private def in_comment?(m)
-          line = position_from_matchdata(m)[:line]
-          processed_source.comments.any? { |c| c.loc.line == line }
+        private def in_comment?(pos)
+          processed_source.comments.any? do |c|
+            c.loc.line == pos[:line] && c.loc.expression.begin_pos <= pos[:column]
```

Comment:
Added the condition:

```
c.loc.expression.begin_pos <= pos[:column]
```


For example, if the source is `foo bar # comment`,

* At what character the search target word starts = position of `foo` ( = 0)
and
* At what character the comment starts = position of `#` (= 8)

are comparing.